sky/utils/common_utils.py

```diff
@@ -456,11 +481,15 @@ def _record(*args, **kwargs):
         return _record
 
 
-def retry(method, max_retries=3, initial_backoff=1):
+def retry(method: Callable[P, R],
+          max_retries: int = 3,
+          initial_backoff: float = 1) -> Callable[P, R]:
     """Retry a function up to max_retries times with backoff between retries."""
 
+    assert max_retries > 0
```

Comment:
We should explicitly raise an exception here.
> Do not use `assert` statements in place of conditionals or validating preconditions. They must not be critical to the application logic. A litmus test would be that the `assert` could be removed without breaking the code. 

https://google.github.io/styleguide/pyguide.html#244-decision

Reply:
> A litmus test would be that the assert could be removed without breaking the code.

So adding this line wont affect the functionality. This function is for dev-only, and we should use assertions here for sanity check.

Reply:
Okay, fair enough.